### PR TITLE
improve autocomplete cycling

### PR DESCRIPTION
**Pull request by stevenctl**
After #10:

1. Type `foo_`
2. (optionally) hit tab to choose a different item
3. Press `<enter>` to select it
4. (optionally) type arguments
5. Press `<enter>` again to execute

After this PR:

1. Type `foo_`, console text is immediately updated with the command
2. (optionally type args)
3. Press `enter` to execute

It's much smoother experience IMO, similar to shells like bash/zsh. 

Also, I removed the left/right arrow key binds because some might want to navigate the text using those keys.

### CODE REVIEW

File: scripts/console_container.gd [hunk 1]
```diff
@@ -117,35 +120,33 @@ func _show_autocomplete(text: String) -> void:
 		_tooltip_label.set_text("\n".join(autocomplete))
 		_tooltip_label.show()
 
-
 func _on_input_text_changed(text: String) -> void:
-	_autocomplete_index = 0
-	_show_autocomplete(text)
-
+	_show_autocomplete(_autocomplete_prompt if _autocomplete_prompt.length() > 0 else text)
```
Comment (4d49) [inline scripts/console_container.gd]: I think it's better that way.
```suggestion
	_show_autocomplete(text if _autocomplete_prompt.is_empty() else _autocomplete_prompt)
```

File: scripts/console_container.gd [hunk 2]
```diff
@@ -117,35 +120,33 @@ func _show_autocomplete(text: String) -> void:
...
 
 func _cycle_autocomplete(direction: int) -> void:
-	var autocomplete: PackedStringArray = _console.autocomplete_list(_console_input.get_text())
+	var autocomplete: PackedStringArray = _console.autocomplete_list(_autocomplete_prompt)
 	_autocomplete_index = wrapi(_autocomplete_index + direction, 0, autocomplete.size())
-
-	_show_autocomplete(_console_input.get_text())
+	_show_autocomplete(_autocomplete_prompt)
```
Comment (4d49) [inline scripts/console_container.gd]: And are you sure it is necessary to call this method? If we call `set_input_text` then inside this method emitted `_console_input.text_changed` which in turn calls `_show_autocomplete`. It looks like we are calling the same method twice. Correct me if I am wrong.
```suggestion
```

File: scripts/console_container.gd [hunk 3]
```diff
@@ -117,35 +120,33 @@ func _show_autocomplete(text: String) -> void:
...
-		if _tooltip_label.is_visible_in_tree():
-			set_input_text(_console.autocomplete_command(_console_input.text, _autocomplete_index))
-		else:
-			_console.execute(_console_input.text)
-			_console_input.clear()
+		_console.execute(_console_input.text)
+		_console_input.clear()
 	elif event.is_action_pressed(&"ui_text_indent"):
+		if _autocomplete_prompt == "":
+			_autocomplete_prompt = _console_input.text
```
Comment (4d49) [inline scripts/console_container.gd]:
```suggestion
		if _autocomplete_prompt.is_empty():
			_autocomplete_prompt = _console_input.text

```

File: scripts/console_container.gd [hunk 4]
```diff
@@ -117,35 +120,33 @@ func _show_autocomplete(text: String) -> void:
...
+	var autocomplete: PackedStringArray = _console.autocomplete_list(_autocomplete_prompt)
 	_autocomplete_index = wrapi(_autocomplete_index + direction, 0, autocomplete.size())
-
-	_show_autocomplete(_console_input.get_text())
+	_show_autocomplete(_autocomplete_prompt)
+	set_input_text(_console.autocomplete_command(_autocomplete_prompt, _autocomplete_index))
 
 func _on_input_gui_event(event: InputEvent) -> void:
+	if event.is_action_type() && !event.is_action(&"ui_text_indent"):
+		# start below zero so the first tab press "selects" the autocomplete
```
Comment (4d49) [inline scripts/console_container.gd]: GDScript style guide [says](https://docs.godotengine.org/en/latest/tutorials/scripting/gdscript/gdscript_styleguide.html#boolean-operators) to use plain english.
```suggestion
	if event.is_action_type() and not event.is_action(&"ui_text_indent"):
		# Start below zero so the first tab press "selects" the autocomplete.
```

File: scripts/console_container.gd [hunk 5]
```diff
@@ -14,7 +14,10 @@ var _console_input : LineEdit
 
 var _tooltip_label : RichTextLabel
 
-var _autocomplete_index: int = 0
+# cycle through the autocomplete options. negative value indicates no selection
+var _autocomplete_index: int = -1
+# preserve the original input text when cycling through options
+var _autocomplete_prompt = ""
```
Comment (4d49) [inline scripts/console_container.gd]:
```suggestion
# Cycle through the autocomplete options. negative value indicates no selection.
var _autocomplete_index: int = -1
# Preserve the original input text when cycling through options.
var _autocomplete_prompt: String = ""
```

**Review by 4d49 — CHANGES_REQUESTED** [review]
A few changes please.

File: scripts/console_container.gd (hunk 2 above)
Comment (stevenctl) [reply scripts/console_container.gd]: You're right. It was needed before, but since we're setting text we can remove it.

**Comment by 4d49** [comment]
Thank you!

**Pull request merged**